From cd1f27ee1be1120387e21d0f5cc7b5fb15ac18ae Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 22 Jun 2020 09:53:55 -0400
Subject: [PATCH] Allow flags other than macros in CPPFLAGS

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8cd3633b877..f3860f42b61 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -651,7 +651,9 @@ ENDFOREACH()
 # Following autotools tradition, add preprocessor definitions
 # specified in environment variable CPPFLAGS
 IF(DEFINED ENV{CPPFLAGS})
-  ADD_DEFINITIONS($ENV{CPPFLAGS})
+  SET(CPPFLAGS $ENV{CPPFLAGS})
+  SEPARATE_ARGUMENTS(CPPFLAGS)
+  ADD_COMPILE_OPTIONS(${CPPFLAGS})
 ENDIF()
 
 INCLUDE(CheckTypeSize)
-- 
2.39.2 (Apple Git-143)

