diff --git a/cmake/boost.cmake.orig b/cmake/boost.cmake
index f807897..8df5b52 100644
--- a/cmake/boost.cmake.orig
+++ b/cmake/boost.cmake
@@ -139,6 +139,7 @@ IF (WITH_BOOST)
             NAMES "${BOOST_PACKAGE_NAME}"
             PATHS ${WITH_BOOST}
             NO_DEFAULT_PATH
+            NO_CMAKE_FIND_ROOT_PATH
            )
   ## Did we get a path name to an unzippped version?
   FIND_PATH(LOCAL_BOOST_DIR
@@ -223,6 +224,7 @@ ENDIF()
 FIND_PATH(BOOST_INCLUDE_DIR
   NAMES boost/version.hpp
   NO_DEFAULT_PATH
+  NO_CMAKE_FIND_ROOT_PATH
   PATHS ${LOCAL_BOOST_DIR}
         ${LOCAL_BOOST_DIR}/${BOOST_PACKAGE_NAME}
         ${WITH_BOOST}
